proxy-server/background/proxy-handler.js

```diff
@@ -0,0 +1,69 @@
+// Log any errors from the proxy script
+browser.proxy.onProxyError.addListener(error => {
+  console.error(`Proxy error: ${error.message}`);
+});
+
+// Initialize the proxy
+function handleInit(message) {
+  // update the proxy whenever stored settings change
+  browser.storage.onChanged.addListener((newSettings) => {
+    if (newSettings.proxyServerInfo) {
+      console.log("Sending updated proxyServerInfo to proxy script");
+      browser.runtime.sendMessage(newSettings.proxyServerInfo.newValue, {toProxyScript: true});
```

Comment:
This is intended as a demonstration of the API. I do not think every possible corner case of potential pitfalls needs to be accounted for. It can convolute the example. I defer to you and Andy, however. Please let me know if you want a "save" button.